### PR TITLE
chore: move all snapshot to insta

### PR DESCRIPTION
**Pull request by hardfist**
## Summary
This PR move all snapshot to insta and we will remove rst and outdated expected folder in the later PR, and when you want to update snapshot do the following commands
```sh
cargo insta test // run snapshot test
cargo insta review // review every changed snapshot
cargo insta accept // accept all new snapshots unconditionally
```
the new snapshot are generated in markdown format so it's easy to read and compare and support syntax highlight( this is inspired by [rome's snapshot](https://github.com/rome/tools/blob/main/crates/rome_js_formatter/tests/specs/jsx/arrow_function.jsx.snap))
we treat .snap as markdown to support highlighting, so it needs configuration for your editor, I already configure it for vscode, if you use other editor you may need to configure yourself if you need snapshot syntax highlight

## Test Plan
all the snapshot moved from expected folder to snapshot folder

### CODE REVIEW

**Comment by Boshen** [comment]
I prefer dumping all the files into a single directory so it is easier to search in your IDEs.

Here is my setup: https://github.com/web-infra-dev/oxc/blob/66e883ae8c3caa0bbe261d6cc48fb424651af89b/crates/oxc_linter/src/tester.rs#L92-L97

**Comment by hardfist** [comment]
> I prefer dumping all the files into a single directory so it is easier to search in your IDEs.
> 
> Here is my setup: [web-infra-dev/oxc@`66e883a`/crates/oxc_linter/src/tester.rs#L92-L97](https://github.com/web-infra-dev/oxc/blob/66e883ae8c3caa0bbe261d6cc48fb424651af89b/crates/oxc_linter/src/tester.rs#L92-L97)

I just follow rome's convention, they put snapshot near the case, but I'm open to change this behavior, I don't use snapshot often so maybe not knowing the tradeoff well, what's your opinions @hyf0 @h-a-n-a @IWANABETHATGUY

**Comment by hyf0** [comment]
I prefer near the case. Just like you would put the build output near the input folder. It will avoid jumping around some folders.

**Comment by hardfist** [comment]
https://github.com/web-infra-dev/rspack/actions/runs/6020195526/job/16331139146?pr=4061#step:6:1002 spot a unstable codegen bug, maybe we need skip it first and fix it in separate PR, cc @underfin

File: crates/rspack/tests/fixtures/simple-with-query/snapshot/output.snap [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+},function(__webpack_require__) {
+var __webpack_exec__ = function(moduleId) { return __webpack_require__(__webpack_require__.s = moduleId) }
+var __webpack_exports__ = (__webpack_exec__("./index.js"));
+
+}
+]);
+```
+```js title=runtime.js
+(function() {
```
Comment (underfin) [inline crates/rspack/tests/fixtures/simple-with-query/snapshot/output.snap:42]: The `runtime.js` shouldn't snapshoted.

Comment (hardfist) [reply crates/rspack/tests/fixtures/simple-with-query/snapshot/output.snap]: removed

**Review by underfin — APPROVED** [review]

**Comment by hardfist** [comment]
https://github.com/web-infra-dev/rspack/actions/runs/6020915278/job/16333122639?pr=4061#step:8:368 the windows ci failed because of other reasons, fix it in other PR

**Pull request merged**